bin/proving-service/src/proxy/status.rs

```diff
@@ -24,3 +37,96 @@ pub async fn create_status_client(
 
     Ok(StatusApiClient::new(channel))
 }
+
+/// Status of a worker
+#[derive(Debug, Serialize, Deserialize)]
+pub struct WorkerStatus {
+    address: String,
+    is_available: bool,
+    is_healthy: bool,
+    retries: usize,
+}
+
+/// Status of the proxy
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ProxyStatus {
+    version: String,
+    supported_proof_types: ProverTypeSupport,
+    total_workers: usize,
+    busy_workers: usize,
+    workers: Vec<WorkerStatus>,
+}
+
+/// Service that handles status requests
+pub struct StatusService {
+    load_balancer: Arc<LoadBalancerState>,
+}
+
+impl StatusService {
+    pub fn new(load_balancer: Arc<LoadBalancerState>) -> Self {
+        Self { load_balancer }
+    }
+
+    async fn handle_request(&self, _session: &mut ServerSession) -> Result<()> {
+        let workers = self.load_balancer.workers.read().await;
+        let worker_statuses: Vec<WorkerStatus> = workers
+            .iter()
+            .map(|w| WorkerStatus {
+                address: w.address(),
+                is_available: w.is_available(),
+                is_healthy: matches!(w.health_status, WorkerHealthStatus::Healthy),
+                retries: w.retries_amount(),
+            })
+            .collect();
+
+        let status = ProxyStatus {
+            version: env!("CARGO_PKG_VERSION").to_string(),
+            supported_proof_types: self.load_balancer.supported_proof_types,
```

Comment:
Related to [this](https://github.com/0xPolygonMiden/miden-base/pull/1277#pullrequestreview-2758713691): Is this decided on startup? If so, could they be conflicting with the worker types?